src/modules/Dashboard/components/CardImage/__tests__/CardImage.test.jsx

```diff
@@ -0,0 +1,130 @@
+import { configure, mount } from "enzyme";
+import Adapter from "enzyme-adapter-react-16";
+import * as React from "react";
+import { Button, Modal } from "react-bootstrap";
+import configureStore from "redux-mock-store";
+import { Provider } from "react-redux";
+import { cleanup } from "@testing-library/react";
+
+import CardImage from "../CardImage";
+import ConfirmDeleteModal from "../../ConfirmDeleteModal/ConfirmDeleteModal";
+import DownloadInfoModal from "../../DownloadInfoModal/DownloadInfoModal";
+import NotesModal from "../../NotesModal/NotesModal";
+
+configure({ adapter: new Adapter() });
+
+afterEach(cleanup);
+
+describe("CardImage", () => {
+    const mockStore = configureStore([]);
+    const deviceList = {
+        "rpi-4-b": {
+            name: "Raspberry Pi",
+            generation: 4,
+            model: "Model B",
+            distros: {
+                "raspbian-buster-armhf": {
+                    name: "Raspbian",
+                    codename: "Buster",
+                    version: 10,
+                    port: "armhf",
+                    packages_url: "https://packages.debian.org/buster/",
+                    build_types: ["classic"],
+                },
+            },
+        },
+    };
+    const image = (status = "Undefined") => ({
+        image_id: "xxxxxxxx-xxxx-4xxx-xxxx-xxxxxxxxxxxx ",
+        device_name: "rpi-4-b",
+        distro_name: "raspbian-buster-armhf",
+        flavour: "Classic",
+        started_at: null,
+        status,
+        notes: "",
+    });
+
+    const render = ({ imageStatus }) => {
+        const initialState = {
+            dashboard: {
+                showNotesSucceededMessage: false,
+            },
+            initialization: {
+                deviceList,
+            },
+        };
+        const store = mockStore(initialState);
+
+        return mount(
+            <Provider store={store}>
+                <CardImage
+                    image={image(imageStatus)}
+                />
+            </Provider>,
+        );
+    };
+
+    /**
+     * Checks the 'href' and 'download' attributes of the download button.
+     */
+    it("Download Button attributes is valid", () => {
+        const component = render({ imageStatus: "Succeeded" });
+        const downloadButton = component.find(Button).at(0);
+        expect(downloadButton.prop("href")).toBe(`/download-image/${image().image_id}.img.gz`);
+        expect(downloadButton.prop("download")).toBe(true);
+    });
+
+    /**
+     * Checks if the modals window for downloading image shown for image in Undefined status.
```

Comment:
Checks if the modal window for downloading an image is shown when the status of the image is Undefined.